cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -184,9 +184,7 @@ func WriteStaticPodToDisk(componentName, manifestDir string, pod v1.Pod) error {
 func GetProbeAddress(cfg *kubeadmapi.MasterConfiguration, componentName string) string {
 	switch {
 	case componentName == kubeadmconstants.KubeAPIServer:
-		if cfg.API.AdvertiseAddress != "" {
-			return cfg.API.AdvertiseAddress
-		}
+		return "127.0.0.1"
```

Comment:
I don't believe we can use DownwardAPI here. Correct me if I am wrong. If we could, I would rather use the `$(HOST_IP)` implemented in https://github.com/kubernetes/kubernetes/commit/49e01b05e7791d8bbb4863f83c0b2f420cec73ab.

Reply:
Maybe I'm missing something but I don't get the argument you made for the initial PR re: HA. 

/cc @jamiehannaford

Reply:
I created a cluster using https://kubernetes.io/docs/setup/independent/high-availability. I also enabled the `SelfHosting` feature. The `DaemonSet` the defines the API server is setup on a "seed" host, which sets up the liveness probes in the pod spec to use `cfg.API.AdvertiseAddress`.  The `cfg.API.AdvertiseAddress` value is automatically set to the seed host's IP. So for masters that are not the seed host, you end up with liveness probes using host that is not their own. Hope that clears it up.